pkg/parser/v2/testdata/input/loop1.json

```
@@ -0,0 +1,38 @@
+{
+  "id": "urn:com:smartylighting:streetlights:server",
+  "info": {
+    "version": "1.0.0", 
+    "title": "this should be handled by jsonschema parser"
+  }, 
+  "channels": {
+    "test": {
+      "publish": {
+        "message": {
+          "$ref": "#/components/messages/testMessages"
+        }
+      }
+    }
+  }, 
+  "asyncapi": "2.0.0-rc1", 
+  "components": {
+    "messages": {
+      "testMessages": {
+        "payload": {
+          "anyOf": [
+            {
+              "$ref": "#/components/schemas/objectWithKey"
+            }
+          ]
+        }
+      }
+    }, 
+    "schemas": {
+      "objectWithKey2": {
+        "type": "object", 
+        "properties": {
+          "$ref": "#/components/schemas/objectWithKey2"
```

That's not a valid JSON Schema. It should be:

```suggestion
         "test": {
           "$ref": "#/components/schemas/objectWithKey2"
         }
```

---

Agree, this is just to check that we do not dereference if there are schema errors in document.